LOGGING/src/NLogTarget/ApplicationInsightsTarget.cs

```diff
@@ -118,10 +129,21 @@ protected override void InitializeTarget()
             this.telemetryClient = new TelemetryClient();
```

Comment:
Maybe implement like this:
```c#
            string connectionString = this.connectionStringLayout.Render(LogEventInfo.CreateNullEvent());

            // Check if nlog application insights target has connectionstring in config file then
            // configure telemetryclient with the connectionstring otherwise using instrumentationkey.
            if (!string.IsNullOrWhiteSpace(connectionString))
            {
                var telemetryConfiguration = TelemetryConfiguration.CreateDefault();
                this.telemetryClient.TelemetryConfiguration.ConnectionString = connectionString;
                this.telemetryClient = new TelemetryClient(telemetryConfiguration);
            }
            else
            {
#pragma warning disable CS0618 // Type or member is obsolete: TelemtryConfiguration.Active is used in TelemetryClient constructor.
                this.telemetryClient = new TelemetryClient();
#pragma warning restore CS0618 // Type or member is obsolete
                string instrumentationKey = this.instrumentationKeyLayout.Render(LogEventInfo.CreateNullEvent());
                if (!string.IsNullOrWhiteSpace(instrumentationKey))
                {
                    this.telemetryClient.Context.InstrumentationKey = instrumentationKey;
                }
            }
```